libcxx/test/std/thread/thread.mutex/thread.lock/thread.lock.guard/assign.compile.pass.cpp

```diff
@@ -14,13 +14,9 @@
 
 #include <mutex>
 
-int main(int, char**)
-{
-    std::mutex m0;
-    std::mutex m1;
-    std::lock_guard<std::mutex> lg0(m0);
-    std::lock_guard<std::mutex> lg(m1);
-    lg = lg0;
+class MyMutex;
 
+int main(int, char**) {
+  static_assert(!std::is_copy_assignable<std::lock_guard<MyMutex> >::value, "");
   return 0;
 }
```

Comment:
```suggestion
static_assert(!std::is_copy_assignable<std::lock_guard<MyMutex> >::value, "");
```

Just as a matter of style, we don't use `main` function in these compile-only tests when they are not needed to avoid any confusion about the test running.

Reply:
Done.